### ISSUE TITLE
Refactor `zeth` and `zeth_debug` commands organization

### ISSUE THREAD

**Issue by AntoineRondelet** [clearmatics/zeth#251]
For now we have to CLIs:
- `zeth`: The main CLI that provides basic functionalities to generate Zeth addresses, interact with the Zeth Mixer etc
- `zeth_debug`: A debug CLI which provides useful functionalities - not strictly related to Zeth - but helping to use Zeth on an Eth-like network etc.

When I run `zeth --help` and `zeth_debug --help` I get:
```bash
(env) > zeth --help
Usage: zeth [OPTIONS] COMMAND [ARGS]...

Options:
  --eth-network TEXT    Ethereum RPC endpoint, network or config file
                        (default: 'eth-network' if it exists, otherwise
                        'ganache')
  --prover-server TEXT  Prover server endpoint (default=localhost:50051)
  --instance-file TEXT  Instance file (default=zeth-instance.json)
  --address-file TEXT   Instance file (default=zeth-address.json)
  --wallet-dir TEXT     Wallet directory (default=./wallet)
  --help                Show this message and exit.

Commands:
  help*          Print help and exit
  deploy         Deploy the zeth contracts and record the instantiation...
  gen-address    Generate a new Zeth secret key and public address
  ls-commits     List all commitments in the Zeth mixer contract
  ls-notes       List the set of notes owned by this wallet
  mix            Generic mix function
  sync           Attempt to retrieve new notes for the key in <key-file>
  token-approve  Approve the mixer to spend some amount of tokens
```

and

```bash
(env) > zeth_debug --help
Usage: zeth_debug [OPTIONS] COMMAND [ARGS]...

Options:
  --eth-network TEXT  Ethereum RPC endpoint, network or config file (default:
                      'eth-network' if it exists, otherwise 'ganache')
  --help              Show this message and exit.

Commands:
  help*                    Print help and exit
  deploy-test-token        Deploy a simple ERC20 token for testing, and
                           mint...
  fund-eth-address         Fund an address.
  generate-eth-address     Locally generate a new Ethereum private key and...
  generate-network-config  Generate a network config file.
  get-eth-balance          Command to get the balance of specific addresses.
```

Some things are not consistent - and/or can be improved here IMO:
1. There is a `token-approve` command in the `zeth` CLI. This is confusing. `zeth` should be about interacting with the Zeth tooling (prover_server, Mixer contract). Moreover, we have a `deploy-test-token` command in `zeth_debug` which is used to deploy the MintableERC20 token. Why would we have "token-related" commands spread across `zeth` and `zeth_debug`?
2. The command `zeth_debug deploy-test-token` does not only - as mentioned in the help message - deploy the token. It also mints it. I would separate these 2 things in 2 different commands for clarity.

Ultimately, here's how I would structure both CLIs:
```
zeth
  |_ deploy
  |_ gen-address
  |_ ls-commits
  |_ ls-notes
  |_ mix
  |_ sync
```

and:

```
zeth_debug
  |_ token
  |    |_ deploy
  |    |_ mint
  |    |_ approve
  |_ eth
  |    |_ gen_address
  |    |_ fund_address
  |    |_ get_balance
  |    |_ gen_network_config
```

**Comment by dtebbs** [comment]
I think the general direction makes sense.  Just a few points we might want to bear in mind while considering the details:

- `deploy-test-token` is for the minimal test token only, so truly only useful for testing. In reality I would expect that this will rarely, if ever, be used outside of our test scripts, where we will need 2 lines instead of 1, but I do see the conceptual point.  

- `token-approve` is intended for the very general case of any token, so we might not want to group it direectly with the test token operations (similarly, grouping `gen_network_config` with the address operations may be worth reconsidering) 

- `token-approve` is also essential for `zeth mix` to succeed. Again, I can see the point about it not being strictly related to `zeth`, but having it only in `zeth_debug` seems like it will just make everything less convenient (I'm treating `zeth_debug` as being something that most users should not need to be aware of, except when experimenting with test deployments). One idea might be to move the command to `zeth_debug`, and *also* expose it as a `--token-approve` flag on the `zeth mix` command?

**Comment by AntoineRondelet** [comment]
Thanks @dtebbs 

1. Yes `deploy-test-token`  deploys OZ's MintableERC20 token. For now, this command uses the hard coded path to this contract, but adding a flag to point to another token can make this command generic enough to be used outside of a testing script IMO.

2. Indeed, if the right allowance is not set on the ERC token, any transferFrom will fail. This is true for any scenario (not only Zeth related). My point is mostly that: if you use an ERC token, you should do what's necessary to use it with Zeth (setting the allowance is not strictly related to Zeth, but rather to the token itself). I do see a point to add a command to set the allowance in the "debug" CLI to make the user's life easier.

To your points above, do I understand correctly that you suggest to drop `gen_network_config` and `deploy-test-token` from the `zeth_debug` cli and keep them as separate scripts? If so, I think that makes sense indeed.

**Comment by dtebbs** [comment]
> To your points above, do I understand correctly that you suggest to drop gen_network_config and deploy-test-token from the zeth_debug cli and keep them as separate scripts? If so, I think that makes sense indeed.

I was not really thinking of them as separate scripts - I think it would be good to have as few extra commands in the env path as possible, keeping all this "non-core" stuff under a single top-level command.  My point was just that not everything may fit neatly into `zeth_debug eth ...` and `zeth_debug token ...` subgroups (if I correctly understood the intention of that structure - I may have been mistaken with that). So we may want to think about that structure a bit.  Or it might be simpler to just keep these 6/7 debug commands as direct children of `zeth_debug`.

**Comment by dtebbs** [comment]
> Yes deploy-test-token deploys OZ's MintableERC20 token. For now, this command uses the hard coded path to this contract, but adding a flag to point to another token can make this command generic enough to be used outside of a testing script IMO.

One reservation I have about adding this kind of functionality relates to all the complexity in handling the include paths and other compiler-related stuff.  Each token may have it's own set of dependencies/includes, compiler flags, compiler versions etc, (and similarly it's own set of deployment parameters). I can't be sure, but I would be concerned that supporting anyting beyond our trivial test token might easily turn into a something beyond the scope of the `zeth` and `zeth_debug` tools.

**Comment by AntoineRondelet** [comment]
Yes definitely get your point. I wasn't proposing a fully generic command taking the path to an arbitrary ERC token (no matter which standard) and deploying it. This would obviously be a rabbit-whole we don't want to get into.
Instead, I was referring to the very simple case where your token contract can be written in different locations (which is not supported now). That being said, your point is valid, and I can't promise this wouldn't trigger a hell if the token we try to compile has some fancy imports - different compiler versions etc. I would assume the compiler would just return an error if the version doesn't match -  but to make all of this very smooth we'd need to extend our config to support multiple compilers (via flags for eg) etc. We don't want to go into that I agree. Let's forget about this token deploy thing (and stick to what we have now)

**Comment by AntoineRondelet** [comment]
Fixed in #254 and #255 
Closing the issue
